streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -411,14 +413,18 @@ void closeSuspended(boolean clean, RuntimeException firstException) {
             if (eosEnabled) {
                 if (!clean) {
                     try {
-                        producer.abortTransaction();
+                        if (!isZombie) {
+                            producer.abortTransaction();
+                        }
                         transactionInFlight = false;
                     } catch (final ProducerFencedException e) {
                         // can be ignored: transaction got already aborted by brokers/transactional-coordinator if this happens
```

Comment:
It's doesn't matter as we are already closing the task here and will not read the value of `transactionInFlight` anymore.